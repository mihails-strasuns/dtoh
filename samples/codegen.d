module codegen;

import std.meta : AliasSeq;

static foreach (int i, T; AliasSeq!(double, int))
    mixin("extern(C) __gshared " ~ T.stringof ~ " global" ~ i.stringof ~ ";");

template T ( )
{
    alias foo_t = extern(C) void function();
}

mixin T;

static if (true)
    extern(C) __gshared foo_t callback;
else
    extern(C) __gshared foo_t no_callback;
