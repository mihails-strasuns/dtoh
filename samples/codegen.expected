#include <stdint.h>

// Used enum definitions:



// Struct forward declaration to resolve cycles:



// Function pointer types:

typedef void (*foo_t)();

// Used struct definitions:



// Variable and function declarations:

double global0;
int32_t global1;
foo_t callback;
